sklearn/tests/test_pipeline.py

```diff
@@ -393,6 +394,38 @@ def test_pipeline_methods_preprocessing_svm():
         pipe.score(X, y)
 
 
+def test_pipeline_methods_pca_tsne():
+    # test that a only fit_transform needs to be present in order to
+    # run a pipeline with fit_transform.
+    # Don't require transform to be present, explicitly.
+    pca = PCA(n_components=2, random_state=0)
+    tsne = TSNE(random_state=0)
+    separate_emb = tsne.fit_transform(pca.fit_transform(iris.data))
+
+    pca_for_pipeline = PCA(n_components=2, random_state=0)
+    tsne_for_pipeline = TSNE(random_state=0)
+    msg = ("Intermediate step '%s' (type %s) does not have "
+           "transform, pipeline is not reusable on test data."
+           % (tsne_for_pipeline, type(tsne_for_pipeline)))
+
+    with pytest.warns(UserWarning, match=re.escape(msg)):
```

Comment:
In most estimators, validation is delayed until fit is called. While this isn't the case for Pipeline, it would be best to check for this warning only when fit_transform is called

Reply:
Do you mean that the check should be delayed until `fit_transform` is called or that the function should be called in the `pytest.warns` scope?

Reply:
I think I mean that the check should be delayed until `fit_transform` is called. But I'm not entirely sure how the latter proposal differs.

Reply:
I initially thought that the call to `self._validate_steps()` should have been removed from `Pipeline.__init__()`.  I've changed the context manager now to guard `fit_transform`